marm-mcp-server/marm_mcp_server/core/memory.py

```
@@ -519,10 +527,20 @@ async def store_memory(self, content: str, session: str, context_type: str = "ge
                 if existing_id:
                     return existing_id
 
+        # Pre-encode once — reused by Layer 2 dedup and storage to avoid a second encode
+        pre_embedding = None
+        pre_embedding_bytes = None
+        if sanitized_content.strip() and self._load_encoder_lazily():
+            try:
+                pre_embedding = await asyncio.to_thread(self._encode_sync, sanitized_content)
+                pre_embedding_bytes = pre_embedding.tobytes()
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)